### PR TITLE
Update multipliers learning rate to scale automatically with dataset size

### LINKED ISSUE #7: Lagrange multipliers need to be manually scaled with the size of the dataset
## Description

Currently we need to use larger multipliers for larger datasets, and smaller multipliers for smaller datasets.

We could potentially just multiply the gradient flowing to each sample from the constraint loss by the number of data points (or simply not divide this loss by the number of data points).

This isn't exactly theoretically sound AFAIK, as the true gradient from FPR or FNR constraints does depend on the number of data points... We could just implement it and see if performance is affected.

### PR DESCRIPTION
**Pull request by AndreFCruz**
Closes #7 and takes precedence over #28.

TODO:
- [x] implement code changes
- [x] test thoroughly (and compare with reference predictions using main branch)
- [ ] implement unit tests to systematize these tests

### CODE REVIEW

File: tests/python_package_test/test_constrained_optimization.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def fairgbm_params(lightgbm_params, target_fpr, target_fnr, target_fairness_metr
     params = lightgbm_params.copy()
     params.update({
         "objective": "constrained_cross_entropy",
-        "multiplier_learning_rate": 10_000,
+        "multiplier_learning_rate": 0.1,
```
Comment (AndreFCruz) [inline tests/python_package_test/test_constrained_optimization.py:41]: this should be 0.05 probably; either that or use a small default tolerance

**Comment by sgpjesus** [comment]
Predictions are an exact match for the code in the main branch after scaling the multipliers with the dataset size:
image
(this is just an eye test to check if the scores are the same) we load the predictions of the model from this branch and compare it to predictions from a FairGBM from the main branch.

image
The results are the same for the two models.
